### PR TITLE
Removing redundant TryGetNonEmptyValue when key is required

### PR DESCRIPTION
**Pull request by eddynaka**
## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] `ReleaseHistory.md` updated for non-trivial changes
* [x] Added unit tests

### CODE REVIEW

**Comment by michaelcfanning** [comment]
Eddy, where is the release note describing this change? No more changes of this kind without documenting them in the release notes, please. :)

Note that this change is actually key to document as we will now aggressively raise exceptions in cases where we previously did not!

**Comment by michaelcfanning** [comment]
>                         Verify(fingerprintText == $"[pat/vs={testCase.Input}]", title, failedTestCases);

this shouldn't work! this is a very old fingerprint. does this test need updating?

---
Refers to: Src/Plugins/Tests.Security/Validators/SEC101_102.AdoPatValidatorTests.cs:76 in c0b866e. [](commit_id = c0b866e3e624e83167fa3b6bbaa5a292475672eb, deletion_comment = False)

**Review by michaelcfanning — CHANGES_REQUESTED** [review]
:clock1:

File: Src/Plugins/Tests.Security/Validators/SEC101_102.AdoPatValidatorTests.cs [hunk 1]
```diff
@@ -53,16 +53,17 @@ public void AzureDevOpsPersonalAccessTokenValidator_IsValidBasic()
                 bool performDynamicValidation = testCase.PerformDynamicValidation;
                 string failureLevel = testCase.FailureLevel;
                 string fingerprintText = null;
-                var groups = new Dictionary<string, FlexMatch>();
+                var groups = new Dictionary<string, FlexMatch>
```
Comment (michaelcfanning) [inline Src/Plugins/Tests.Security/Validators/SEC101_102.AdoPatValidatorTests.cs:56]: >groups [](http://example.com/codeflow?start=20&length=6)

why didn't we notice this problem before? looks like we're perhaps missing a key unit test somewhere?

Comment (eddynaka) [reply Src/Plugins/Tests.Security/Validators/SEC101_102.AdoPatValidatorTests.cs:45]: This was a strange test.
I removed a few things and started validating the expected count and state.

**Comment by eddynaka** [comment]
>                         Verify(fingerprintText == $"[pat/vs={testCase.Input}]", title, failedTestCases);

This isn't needed. just removed.

---
In reply to: [899106459](https://github.com/microsoft/sarif-pattern-matcher/pull/539#issuecomment-899106459) [](http://example.com/codeflow?ancestors=899106459,899106459)

---
Refers to: Src/Plugins/Tests.Security/Validators/SEC101_102.AdoPatValidatorTests.cs:76 in c0b866e. [](commit_id = c0b866e3e624e83167fa3b6bbaa5a292475672eb, deletion_comment = False)

**Comment by eddynaka** [comment]
Added a new entry

---
In reply to: [899106233](https://github.com/microsoft/sarif-pattern-matcher/pull/539#issuecomment-899106233) [](http://example.com/codeflow?ancestors=899106233,899106233)

File: Src/ReleaseHistory.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+- Plugin Improvement: Required properties will throw exception if it does not exist. [#539](https://github.com/microsoft/sarif-pattern-matcher/pull/539)
```
Comment (michaelcfanning) [inline Src/ReleaseHistory.md:5]: >it does [](http://example.com/codeflow?start=66&length=7)

'if they do not exist'.

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**